lib/optimize/ConcatenatedModule.js

```diff
@@ -1241,7 +1241,10 @@ class ConcatenatedModule extends Module {
 								const path = getPathInAst(info.ast, identifier);
 								if (path && path.length > 1) {
 									const maybeProperty =
-										path[1].type === "AssignmentPattern" ? path[2] : path[1];
+										path[1].type === "AssignmentPattern" &&
+										path[1].left === path[0]
```

Comment:
I thought about writing this as `path[1].left === identifier`. Exactly the same thing, but perhaps the intent would be clearer?